src/Uno.UI/UI/Xaml/Controls/TextBox/SinglelineTextBoxDelegate.iOS.cs

```diff
@@ -58,27 +60,38 @@ public override bool ShouldChangeCharacters(UITextField textField, NSRange range
 
 		public override bool ShouldReturn(UITextField textField)
 		{
-			if (IsKeyboardHiddenOnEnter)
+			try
 			{
-				_ = CoreDispatcher.Main.RunAsync(CoreDispatcherPriority.Normal,
-					async () =>
+				if (IsKeyboardHiddenOnEnter)
+				{
+					_ = CoreDispatcher.Main.RunAsync(CoreDispatcherPriority.Normal,
+						async () =>
+						{
+							// Delay losing focus to avoid concurrent interactions when transferring focus to another control. See 101152
+							await Task.Delay(TimeSpan.FromMilliseconds(50));
+							textField.ResignFirstResponder();
+						});
+				}
+
+				if (textField is SinglelineTextBoxView textBox)
+				{
+					if (_textBox.GetTarget()?.OnKey('\n') ?? false)
 					{
-						// Delay losing focus to avoid concurrent interactions when transferring focus to another control. See 101152
-						await Task.Delay(TimeSpan.FromMilliseconds(50));
-						textField.ResignFirstResponder();
-					});
-			}
+						return false;
+					}
+				}
 
-			var textBox = textField as SinglelineTextBoxView;
-			if (textBox != null)
+				return true;
+			}
+			catch (Exception ex)
 			{
-				if (_textBox.GetTarget()?.OnKey('\n') ?? false)
+				if (this.Log().IsEnabled(LogLevel.Warning))
```

Comment:
I think this deserves to have `Error` level

Reply:
Copied!